### DISCUSSION TITLE
MotionPlus and Odroid arm64

### DISCUSSION THREAD

**Discussion by gtbyio** [Motion-Project/motionplus#50] [General]
Hello

a request for a helping hand, how to use the script for an Arm64 (Odroid-C2)
or another solution to install MotionPlus.

https://motion-project.github.io/motionplus_build.html

with the script I am looking for what information given for branch, install, arch.
 
an example:

`EBUSERNAME=$1
DEBUSEREMAIL=$2
GITBRANCH=$3
INSTALLPKG=$4
ARCH=$5`   

 ARCH=$5 ???  = arm64??

yes for motionplus to develop it is necessary to put at the disposal of the world more detailed information, the creation of package for several architecture would be a good thing.

a .deb for arm64 !! yes

**Comment by Mr-Dave** [comment]
I recommend you just leave it blank and run `./buildplus.sh`.  The master branch is the default which is the best choice, no is the default for install packages (so it will list what to install), and the Arch is a cross compiling item.

**Comment by gtbyio** [comment]
Thank you very much Mister Dave for the quick answer, I will test it and let you know.

**Comment by gtbyio** [comment]
Bravo for the script, functional at 200%.
details of the creation of the Deb package (this can be useful for others)

on a base :  ODROID C2  /  Armbian 23.02.2 Focal with Linux 6.1.11-meson64

#download script
odroidc2a:~$ wget   ......otion-Project/motion-packaging/master/buildplus.sh
odroidc2a:~$ sudo chmod +x buildplus.sh

#creation of the package 
odroidc2a:~$ sudo bash buildplus.sh
odroidc2a:~$  ls  (list of created files )
```
focal_motionplus_0.1.0-git-20230325-a5d594f-1.dsc
focal_motionplus_0.1.0-git-20230325-a5d594f-1_arm64.buildinfo  focal_motionplus_0.1.0-git-20230325-a5d594f-buildlog-aarch64.txt
focal_motionplus_0.1.0-git-20230325-a5d594f-1_arm64.changes    focal_motionplus_0.1.0-git-20230325-a5d594f.orig.tar.gz
focal_motionplus_0.1.0-git-20230325-a5d594f-1_arm64.deb        influxdata-archive_compat.key
focal_motionplus_0.1.0-git-20230325-a5d594f-1.debian.tar.xz
```
odroidc2a:~$ sudo apt-get install  pkgconf libtool libavcodec-dev libavdevice-dev libavformat-dev libswscale-dev libjpeg-dev libpq-dev libsqlite3-dev debhelper dh-autoreconf zlib1g-dev libwebp-dev libmicrohttpd-dev gettext fakeroot libasound2-dev libpulse-dev libfftw3-dev libopencv-dev libmariadb-dev


# installs deb package
odroidc2a:~$  sudo dpkg -i focal_motionplus_0.1.0-git-20230325-a5d594f-1_arm64.deb

#terminal view  :
`.....Adding group `motion' (GID 120) ... Done.
Adding system user `motion' (UID 113) ...
Adding new user `motion' (UID 113) with group `motion' ...
Not creating home directory `/var/lib/motionplus'.
Adding user `motion' to group `video' ...
Adding user motion to group video
Done.
Created symlink /etc/systemd/system/multi-user.target.wants/motionplus.service → /lib/systemd/system/motionplus.service.`

..... Remains to do the configuration of the file 

(I don't know where to put the .deb package for the users?  )

                                                  ---------------------   ***  ----------------------------

now I open a new thread for webcam configuration 

https://v1.casimages.com/v/2023/04/02//prVIPb-vokoscreen-2023-04-02-10-38-59.mp4
